### PR TITLE
Initial version of CODEOWNERS

### PR DESCRIPTION
**Pull request by pawelchcki**
# What does this PR do?

Attempt at a first iteration of CODEOWNERS file. Created as a Separate PR to #50 to verify its working as expected, and iterate on the ownership separately

# Motivation

CODEOWNERS -  can -  help streamline the review process. Maybe

### CODE REVIEW

**Comment by ivoanjo** [comment]
I'm a bit of a noob around `CODEOWNERS` -- what advantages does setting it up get us vs just having a flat model?

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,4 @@
+*               @Datadog/libdatadog-core
+ddprof*/        @Datadog/profiling
```
Comment (r1viollet) [inline .github/CODEOWNERS:2]: only library people are relevant, though do you want to create a new group or just list the people here ?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+ddtelemetry/    @Datadog/instrumentations-reliability
+ruby/           @ivoanjo
```
[the remark this answers is not in the document]
Comment (ivoanjo) [reply .github/CODEOWNERS:4]: Did you mean...

Time to find _**even more**_ awesome ruby devs? :P

File: .github/CODEOWNERS (hunk 1 above)
Comment (pawelchcki) [reply .github/CODEOWNERS]: Good point. I've added new team libdatadog-profiling and added you all as maintainers.

**Comment by pawelchcki** [comment]
>I'm a bit of a noob around CODEOWNERS -- what advantages does setting it up get us vs just having a flat model?
@ivoanjo - it mostly just streamlines the review, especially for newcomers to the repo

**Review by morrisonlevi — APPROVED** [review]

**Pull request merged**